anghammarad-client-node/test/main.test.ts

```diff
@@ -0,0 +1,66 @@
+import {Anghammarad} from "../src/main";
+import {NotifyParams, RequestedChannel} from "../src/interfaces";
+
+describe("The messageJson function", () => {
+  // Pass something in here so that we don't bother instantiating a client
+  // each time we run the tests
+  const client = new Anghammarad({} as any);
+
+  const defaultParams: NotifyParams = {
+    subject: "subject",
+    message: "message",
+    actions: [{ cta: "cta", url: "url" }],
+    target: { Stack: "stack" },
+    channel: RequestedChannel.All,
+    sourceSystem: "source",
+    topicArn: "arn",
+  };
+
+  it("sets message as provided", () => {
+    expect(client.messageJson(defaultParams)).toContain(`"message":"message"`);
+  });
+
+  it("sets sender as provided", () =>{
+    expect(client.messageJson(defaultParams)).toContain(`"sender":"source"`)
+  })
+
+  describe("represents channel correctly", () => {
+    it("for 'All'", () => {
+      expect(client.messageJson(defaultParams)).toContain(`"channel":"all"`)
+    })
+
+    it("for 'Email'", () => {
+      expect(client.messageJson({...defaultParams, channel: RequestedChannel.Email})).toContain(`"channel":"email"`)
+    })
+
+    it("for 'Hangouts'", () => {
+      expect(client.messageJson({...defaultParams, channel: RequestedChannel.HangoutsChat})).toContain(`"channel":"hangouts"`)
+    })
+
+    it("for 'Prefer Email'", () => {
+      expect(client.messageJson({...defaultParams, channel: RequestedChannel.PreferEmail})).toContain(`"channel":"prefer email"`)
+    })
+
+    it("for 'Prefer Hangouts'", () => {
+      expect(client.messageJson({...defaultParams, channel: RequestedChannel.PreferHangouts})).toContain(`"channel":"prefer hangouts"`)
+    })
+  })
+
+  it("includes target", () => {
+    const result = client.messageJson({...defaultParams, target: {Stack: "stack-name", App: "app-name"}})
+    expect(result).toContain(`"Stack":"stack-name"`)
+    expect(result).toContain(`"App":"app-name"`)
+  })
+
+  it("includes actions", () => {
+    const result = client.messageJson({...defaultParams, actions: [{cta: "cta1", url: "url1"}, {cta: "cta2", url: "url2"}]})
+    expect(result).toContain(`"cta":"cta1"`)
+    expect(result).toContain(`"url":"url1"`)
+    expect(result).toContain(`"cta":"cta2"`)
+    expect(result).toContain(`"url":"url2"`)
+  })
+
+  it("properly escapes input", () => {
+    expect(client.messageJson({...defaultParams, message: `Message with "quotes"`})).toContain(`\"`)
```

Comment:
I'm not crazy about this assertion though. As above, it might be better to parse the JSON and check the value contains quotes. I'd also observe that this is really two "properties", so this could be two separate tests:
* the resulting JSON is valid when quotes appear in the data
* the field containing quotes is correctly encoded

This test will erroneously pass in lots of ways depending on the test data provided and the implementation itself. Someone that comes to the codebase later and changes either of those things may not know that they need to update this test accordingly. If the test is more focused on the properties being enforced (that the resulting JSON is valid, and that the message field contains the correct data) then we don't have to rely on assumptions elsewhere in the codebase for this test to be valuable.

Reply:
Similar to the above comment, we're just testing that `JSON.stringify` knows what to do with something like 
```
`I'm a string with "quotes" in`
```